src/audio/eq_fir.c

```diff
@@ -175,7 +175,7 @@ static void eq_fir_s16_passthrough(struct fir_state_32x16 fir[],
 	int16_t *dest = (int16_t *)sink->w_ptr;
 	int n = frames * nch;
 
-	memcpy(dest, src, n * sizeof(int16_t));
+	memcpy_s(dest, n * sizeof(int16_t), src, n * sizeof(int16_t));
```

Comment:
This should rather check the distance between r_ptr/w_ptr and their respective end of buffers.